### PR TITLE
Fixed Gradle  issues and Minify rules issues

### PR DESCRIPTION
**Pull request by hasanm08**
Gradle upgraded to 8
java Upgraded to 17
androidR8 build issue fixed
minify rules issues fixed

### CODE REVIEW

**Comment by Ansh934** [comment]
Does your PR solve issue #230 ?

**Comment by hasanm08** [comment]
> Does your PR solve issue #230 ? 

Yes
By changing proguard file it will be solved
Check
example/android/app/proguard-rules.pro

File: example/ios/Flutter/flutter_export_environment.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+# This is a generated file; do not edit or check into version control.
```
Comment (jaumard) [inline example/ios/Flutter/flutter_export_environment.sh:2]: shouldn't be commit, can you remove it please?

**Comment by hasanm08** [comment]
@jaumard minor issues fixed please merge

**Review by jaumard — APPROVED** [review]

**Comment by jaumard** [comment]
Thanks for this!

**Pull request merged**